### DISCUSSION TITLE
Clarification needed on transmission and refraction

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3702] [Q&A] [answered]
A colleague and I were unclear on the meaning of the sentence "The transmission property defines what ratio of diffuse light is transmitted through a refractive material."  Does this only affect reflected light that's coming through the translucent material and reaching the eye, or could it affect light from a light source that passes through the translucent material and then reflects off something else?  For example, could you put a thin translucent red filter in front of a white light source and cause surrounding objects to look red?  The latter would be cool, but I doubt that it could be done without ray tracing.

**Comment by romainguy** [comment, marked as answer]
This effect only filters the appearance of the refractive object itself, it does not change the light rays. You are indeed right that doing so would require a ray tracer :(

**Comment by hillesland** [comment]
If the film is really thin, you don't need raytracing.   You do something like shadow mapping, but instead of. removing the light, you modulate it.  Although it doesn't scale well beyond one layer of film.
